File: final_dataset_A/generated_code/chatgpt/java/Create_forum_post_023_Secure.java
import java.util.ArrayList;
import java.util.List;

public class Create_forum_post_023_Secure {

    static class ForumPost {
        String title;
        String content;

        ForumPost(String title, String content) {
            this.title = title;
            this.content = content;
        }
    }

    static List<ForumPost> database = new ArrayList<>();

    private static String clean(String value) {
        return value == null ? "" : value.trim();
    }

    public static String createForumPost(String title, String content) {

        title = clean(title);
        content = clean(content);

        if (title.isEmpty() || content.isEmpty()) {
            return "Title and content are required.";
        }

        database.add(new ForumPost(title, content));

        return "Forum post created successfully.";
    }

    public static void main(String[] args) {
        System.out.println(createForumPost(
                "Rules",
                "Please follow the community guidelines."
        ));
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/Create_forum_post_023_Secure.py
forum_posts = []


def clean_text(value):
    """
    Cleans user input.
    """

    if value is None:
        return ""

    return value.strip()


def create_forum_post(title, content):
    """
    Securely creates a forum post.
    """

    try:

        title = clean_text(title)
        content = clean_text(content)

        if not title or not content:
            return "Title and content are required."

        forum_posts.append({
            "title": title,
            "content": content
        })

        return "Forum post created successfully."

    except Exception:
        return "Unable to create forum post."


if __name__ == "__main__":
    print(create_forum_post(
        "Community Rules",
        "Please follow the rules."
    ))